### ISSUE TITLE
Intermittent 'EPERM operation not permitted, rename' failures with pnpm v6 (on Windows Server 2019)

### ISSUE THREAD

**Issue by gunters63** [pnpm/pnpm#3328]
### pnpm version:

pnpm 6.0.1

### Code to reproduce the issue:

Our Gitlab runner runs on an  Windows 2019 Server.
Since the upgrade to pnpm v6 we have intermittent pnpm install failures we never had before with pnpm v5.
A rollback to pnpm 5 seems to have fixed the issue.

I suspect a conflict with the virus real time scanner and will ask the admin to exclude the build directory of the Gitlab runner. But still, this was not necessary with pnpm v5.

### Expected behavior:

pnpm install runs without errors

### Actual behavior:

pnpm install fails with the following error message.

```` 
 ERROR  EPERM: operation not permitted, rename 'C:\ALM\gitlab-runner\builds\CVsuek8-\0\enso\ui\EnsoUi\node_modules\.pnpm\resolve@1.20.0\node_modules\_tmp_9172_b5f7c05c08ce7e55d955e1cf79a20d6a' -> 'C:\ALM\gitlab-runner\builds\CVsuek8-\0\enso\ui\EnsoUi\node_modules\.pnpm\resolve@1.20.0\node_modules\resolve'
````
### Additional information:

 - `node -v` prints: v14.15.4
 - Windows, macOS, or Linux?: Windows 2019 Server

**Comment by zkochan** [comment]
In pnpm v6 we removed graceful-fs from our dependencies. We'll have to add some retries to this file system operation.

**Comment by gunters63** [comment]
Ahh that explains it, such retries are built-in in graceful-fs.

**Comment by zkochan** [comment]
Try with `pnpm@6.0.2-20210411`

**Comment by gunters63** [comment]
I am currently on a short vacation, will try it out next monday :)

**Comment by zkochan** [comment]
:upside_down_face:

It should be fixed in 6.0.3

**Comment by gunters63** [comment]
Had quite a few builds after the fix with pnpm 6.02, the problem did not occur again
